### ISSUE TITLE
Multiple runtimes support 

### ISSUE THREAD

**Issue by harche** [containerd/cri-containerd#623]
Hi,

Does CRI-Containerd support multiple OCI runtimes support like [CRI-O](https://github.com/kubernetes-incubator/cri-o/blob/6faef132934e84fb8c1427f9c7415a60de519321/cmd/crio/config.go#L57)? 

If it doesn't I can rename this issue to RFC and we can discuss how can be have it in CRI-Containerd.

**Comment by mikebrow** [comment]
You can try it out see the "--containerd-runtime-engine" option. "runc" is the default... Any particular runc alternatives you are thinking about?

**Comment by mikebrow** [comment]
If you do and have problems, report and I'll take a look.  Cheers!

**Comment by Random-Liu** [comment]
@harche What do you mean by `support multiple OCI runtimes`?

If you are talking about running cri-containerd/containerd with a different OCI runtime, you can use the flag mentioned by @mikebrow.

If you want to run cri-containerd/containerd with multiple different OCI runtimes at a time, we don't support it for now. The main reason is that Kubernetes and CRI hasn't defined this kind of thing. :)

Which one are you talking about? And what is your use case? I'm curious. @harche

**Comment by harche** [comment]
@mikebrow @Random-Liu 

We are thinking of using upcoming [kata containers](https://katacontainers.io/) for some of the security sensitive workloads to be run in greater isolation. 

Considering large configuration of the boxes that we ship, we would really love to have a functionality to run 2 runtimes simultaneously. That way customers don't have a dedicate an entire system just to one type of workload. They should be able to choose to run regular containers (using `runc`) while run workloads they deem sensitive in higher isolation (using something like kata) on the same system.

CRI-Containerd can take [similar approach](https://github.com/kubernetes-incubator/cri-o/blob/6faef132934e84fb8c1427f9c7415a60de519321/cmd/crio/config.go#L57)

**Comment by dungeonmaster18** [comment]
As far as katacontainers support is concerned, it is there as a [gsoc proposal](https://github.com/cncf/soc/blob/master/README.md#katacontainers-support-for-containerdcri-containerd) and so I hope it is going to be there very soon.

**Comment by chengyli** [comment]
Looks that cri-containerd has support both runc and kata-runtime now.

```
    [plugins.cri.containerd]
      snapshotter = "overlayfs"
      no_pivot = false
      [plugins.cri.containerd.default_runtime]
        runtime_type = "io.containerd.runtime.v1.linux"
        runtime_engine = "/usr/local/sbin/runc"
      [plugins.cri.containerd.untrusted_workload_runtime]
        runtime_type = "io.containerd.runtime.v1.linux"
        runtime_engine = "/usr/bin/kata-runtime"
```

**Comment by resouer** [comment]
@harche @chengyli Yes exactly!

We just cooperated with containerd folks and implemented cri-containerd + katacontiners by latest shimv2 api which works like charming. The PoC has already passed ALL node e2e tests.

You can track this progress here: https://github.com/kata-containers/runtime/pull/572 in which containerd + kata integration will become official part of kata runtime.

And you can also play with this develop branch as well https://github.com/hyperhq/kata-runtime/tree/shimv2

**Comment by egernst** [comment]
Well, to clarify - it supports kata-runtime now.  It's just that containerd + kata work much cleaner and more efficiently with https://github.com/kata-containers/runtime/pull/572

**Comment by harche** [comment]
Yeah, I have been using kata with containerd (with custom containerd config) from quite some time now.
